File: dsa/go/bubbleSort.go
package main

import (
	"fmt"
)

func main() {
	list := []int{23, 1, 56, 90, 56, 43, 75}
	newList := bubbleSort(list)
	for _, v := range newList {
		fmt.Println(v)
	}
}

func bubbleSort(list []int) []int {
	for i := 0; i < len(list); i++ {
		swapped := false
		for u := 0; u < len(list)-1; u++ {
			if list[u] > list[u+1] {
				list[u], list[u+1] = list[u+1], list[u]
				swapped = true
			}
		}

		if !swapped {
			break
		}
	}

	return list
}

File: dsa/typescript/bubbleSort.js
var list = [22, 18, 87, 13, 1, 89, 43, 51, 77, 93];
var newList = bubbleSort(list);
console.log(newList);
function bubbleSort(list) {
    for (var i = 1; i < list.length; i++) {
        var swapped = false;
        for (var u = 0; u < list.length - 1; u++) {
            if (list[u] > list[u + 1]) {
                var temp = list[u];
                list[u] = list[u + 1];
                list[u + 1] = temp;
                swapped = true;
            }
        }
        if (!swapped) {
            break;
        }
    }
    return list;
}
